### PR TITLE
RunRequests support asset_selection

### PR DESCRIPTION
**Pull request by OwenKephart**
### Summary & Motivation

It's nice to be able to kick off just a selection of assets on a job rather than the whole thing

### CODE REVIEW

File: python_modules/dagster/dagster/_core/definitions/run_request.py [hunk 1]
```diff
@@ -72,13 +76,17 @@ def __new__(
         run_config: Optional[Mapping[str, Any]] = None,
         tags: Optional[Mapping[str, str]] = None,
         job_name: Optional[str] = None,
+        asset_selection: Optional[List[AssetKey]] = None,
```
Comment (sryza) [inline python_modules/dagster/dagster/_core/definitions/run_request.py:79]: Nitpick: Sequence

**Review by sryza — APPROVED** [review]
Oh wow very simple. Is there any reason we might want this to be experimental?

**Comment by OwenKephart** [comment]
@sryza One thing to think about is if we should allow the `asset_selection` argument to take in an `AssetSelection` object (instead of a list of asset keys). Also there's a bit of inconsistency around the codebase on if asset_selection is a Set[AssetKey] or List[AssetKey], so we might want to give ourself some leeway there (although realistically we should probably just accept either).

cc @clairelin135 if you have thoughts

**Comment by clairelin135** [comment]
I think we should definitely allow `AssetSelection` objects in asset_selection arguments. In a sensor, you would ideally want to kick off materializations for an asset an all of its downstream assets. It's way more ergonomic to be able to say `AssetSelection("a").downstream()` than to have to manually specify all the asset keys.

**Comment by sryza** [comment]
@clairelin135 good point

**Pull request merged**